### ISSUE TITLE
Collected issues from Akamai ingest team review

### ISSUE THREAD

**Issue by wilaw** [Dash-Industry-Forum/Ingest#75]
Various architects [thank you Yiliang Bao and Vikram Pentakota] within Akamai reviewed the spec on May 23rd and these comments have been collected here . 

**Section 5 General Ingest Protocol Behavior**
1. Suggest to add that Ingest Source SHALL support HTTP 3XX redirects.
2. Suggest to add that the ingest source and receiving entity SHALL support posting an object using "Content-Encoding: gzip".
3. [5.5]  In addition to requiring basic authentication HTTP AUTH [RFC7617], suggest to add "or HTTP Digest Access Authentication RFC 7616"
4. [5.6] conflicts with [5.5] which requires one of the listed options. [5.6] makes client cert (TLS mutual auth) mandatory. "self assigned" or "self signed"? If a self signed certificate is used, does using client cert really improve security?
5. Suggest change [5.10] "terminating" to "terminating connection"
6. [5.12] Remove "transport" as IP layer is not transport layer. Change the text to: "Ingest source MUST support IPv4 and IPv6. Receiving entity MUST support IPv4, and it SHOULD support IPv6."
7. in [5.14]. with respect to " ingest can resume sending ", should it be "ingest source" instead of "ingest"?
8. [5.16] - Should clarify this is ingest source behavior.
9. [5.17] - should return 403 with error reason.
10. [5.20] - "Otherwise" is ambiguous. Not sure what it means here?
11. [5.21] - The receiving entity MAY return 4xx or 5xx HTTP error code in case ...

**Section 7 Ingest Interface 2: DASH and HLS Ingest Protocol Behavior** 

12. [7.1.1] Should support for HTTP/2.0 be required?
13. [7.1.2.4] - "a different TCP connection" instead of "a different TCP sesesion"?
14. [7.1.3.6] - suggest that init segments can be specified either by the .init file extension OR the string "init" in the filename. 
15. [Table 6] - add proper mime type for gzipped playlist
16. [7.1.5.1] - Change SHOULD to SHALL. The Ingest source SHALL include a User-Agent header. 
17 [7.1.6.1b] - when a discontinuity marker is inserted, does it mean the slow segment is removed from playlist? Will discontinuity marker be inserted into variant playlist of other renditions?
18. [7.1.6.1b] - May add use case to explain why a slow segment is still needed even discontinuity marker is inserted into the playlist . For example, an event playlist will be uploaded at the end of presentation, so we have complete list of segments.
19. [7.2.1.1] - suggest master and variant playlist, instead of parent and child playlist.
20. [7.2.1.2] - Do we have to use ".key" extension for key file? This should be optional.
21. [7.2.1.3] - MPEG Transport Stream (TS) instead of Transport Stream.
22. [7.2.1.4] - suggest improve language to make it clear that MPEG-CMAF content should not use a .ts file extension, but that content encapsulated in a Transport Stream file container MUST use a .ts extension. 
23. [7.2.4.1] - URL of variant playlist in master playlist SHOULD also use relative URL.
24. [7.2.5.1] - Is this just file name or does it includes partial URL path?
25. [7.2.5.2] - How is "restart object names" different from "use previously used object names"?

**Comment by RufaelDev** [comment]
Thank you for valuable comments, I updated the spec text accordingly see inline, also i created two new issues. 

Suggest to add that Ingest Source SHALL support HTTP 3XX redirects.
    [RM] what should the ingest source do with this redirect message ? 
Suggest to add that the ingest source and receiving entity SHALL support posting an object using "Content-Encoding: gzip".
    [RM]  accept
[5.5] In addition to requiring basic authentication HTTP AUTH [RFC7617], suggest to add "or HTTP Digest Access Authentication RFC 7616" 
     [RM] to be discussed in next call created issue #76
[5.6] conflicts with [5.5] which requires one of the listed options. [5.6] makes client cert (TLS mutual auth) mandatory. "self assigned" or "self signed"? If a self signed certificate is used, does using client cert really improve security?
    [RM] improved this, self assigned mainly for when the sender and receiver are under the same domain administration 
Suggest change [5.10] "terminating" to "terminating connection"
   [RM] accepted
[5.12] Remove "transport" as IP layer is not transport layer. Change the text to: "Ingest source MUST support IPv4 and IPv6. Receiving entity MUST support IPv4, and it SHOULD support IPv6."
    [RM] accepted
in [5.14]. with respect to " ingest can resume sending ", should it be "ingest source" instead of "ingest"?
 [RM] accept 
[5.16] - Should clarify this is ingest source behavior.
 [RM] accepted 
[5.17] - should return 403 with error reason. 
[RM] accepted, how should the error reason be communicated ?
[5.20] - "Otherwise" is ambiguous. Not sure what it means here?
[RM] updated, some receivers may not be able to reply 412 
[5.21] - The receiving entity MAY return 4xx or 5xx HTTP error code in case ...
Section 7 Ingest Interface 2: DASH and HLS Ingest Protocol Behavior
[7.1.1] Should support for HTTP/2.0 be required?
[RM] not yet, in next version it can be added
[7.1.2.4] - "a different TCP connection" instead of "a different TCP sesesion"?
[RM] accepted
[7.1.3.6] - suggest that init segments can be specified either by the .init file extension OR the string "init" in the filename.
[RM] this was discussed, having init in the file name may give conflicts when the name also includes it, this was removed from the spec
[Table 6] - add proper mime type for gzipped playlist
[RM] accept
[7.1.5.1] - Change SHOULD to SHALL. The Ingest source SHALL include a User-Agent header.
[RM] does this really affect the interop ? is this header not used for debugging mostly ? Should can be sufficient in this case
17 [7.1.6.1b] - when a discontinuity marker is inserted, does it mean the slow segment is removed from playlist? Will discontinuity marker be inserted into variant playlist of other renditions?
[RM]yes, but segment may be available still in other renditions
[7.1.6.1b] - May add use case to explain why a slow segment is still needed even discontinuity marker is inserted into the playlist . For example, an event playlist will be uploaded at the end of presentation, so we have complete list of segments.
[RM] added the archiving use case
[7.2.1.1] - suggest master and variant playlist, instead of parent and child playlist.
[7.2.1.2] - Do we have to use ".key" extension for key file? This should be optional.
[RM] I understood the naming was important for this profile, but i made it optional
[7.2.1.3] - MPEG Transport Stream (TS) instead of Transport Stream. 
[RM] accepted 
[7.2.1.4] - suggest improve language to make it clear that MPEG-CMAF content should not use a .ts file extension, but that content encapsulated in a Transport Stream file container MUST use a .ts extension
[RM] accepted.
[7.2.4.1] - URL of variant playlist in master playlist SHOULD also use relative URL.
[RM] accepted
[7.2.5.1] - Is this just file name or does it includes partial URL path?
RM filename
[7.2.5.2] - How is "restart object names" different from "use previously used object names"?
[RM] removed the sentence as this was contradictory, not using different object names for the same content is a requirement that is already covered by the uniqueness
